src/main/webapp/app/exam/participate/exam-participation.component.ts

```diff
@@ -159,7 +158,7 @@ export class ExamParticipationComponent implements OnInit, OnDestroy, ComponentC
     // displays the alert for confirming leaving the page if there are unsaved changes
     @HostListener('window:beforeunload', ['$event'])
     unloadNotification($event: any): void {
-        if (!this.isOver()) {
+        if (!this.isOver() && this.studentExam != null && !this.handInEarly) {
```

Comment:
I'm not sure, but can't we just use `!canDeactivate()` here? That would remove some code duplication.

Reply:
yes you are right, good catch 👍